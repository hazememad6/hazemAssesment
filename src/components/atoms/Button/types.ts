import { TouchableOpacityProps } from "react-native";

export interface ButtonProps extends Omit<TouchableOpacityProps, "onPress"> {
  title?: string;
  onPress: () => void;
  variant?: "primary" | "secondary" | "outline" | "ghost";
  size?: "small" | "medium" | "large";
  disabled?: boolean;
  loading?: boolean;
  icon?: string;
  iconOnly?: boolean;
}
