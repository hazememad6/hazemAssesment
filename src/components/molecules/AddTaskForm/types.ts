export interface AddTaskFormProps {
  onSubmit: (task: { title: string; description?: string; completed: boolean }) => void;
  loading?: boolean;
}
