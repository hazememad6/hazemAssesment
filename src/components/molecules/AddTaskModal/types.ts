import { Task } from "src/types/task";

export interface AddTaskFormData {
  title: string;
  description?: string;
}

export interface AddTaskModalProps {
  visible: boolean;
  onClose: () => void;
  onAddTask: (task: Omit<Task, "id" | "createdAt" | "updatedAt">) => void;
  loading?: boolean;
}
