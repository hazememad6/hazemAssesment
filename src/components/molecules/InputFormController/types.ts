import { Control, FieldPath, FieldValues, RegisterOptions } from "react-hook-form";
import { InputProps } from "@components/atoms/Input/types";

export interface InputFormControllerProps<
  TFieldValues extends FieldValues = FieldValues,
  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
> extends Omit<InputProps, "value" | "onChangeText" | "onBlur" | "error"> {
  control: Control<TFieldValues>;
  name: TName;
  rules?: RegisterOptions<TFieldValues, TName>;
}
