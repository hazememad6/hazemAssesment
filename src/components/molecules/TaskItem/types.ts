import { Task } from "src/types/task";

export interface TaskItemProps {
  task: Task;
  onComplete: (id: string) => void;
  onDelete: (id: string) => void;
  isUpdating?: boolean; // loading state for toggle
  isDeleting?: boolean; // loading state for delete
}
