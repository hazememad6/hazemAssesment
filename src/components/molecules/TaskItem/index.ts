export { TaskItem } from "./TaskItem";
export type { TaskItemProps } from "./types";
