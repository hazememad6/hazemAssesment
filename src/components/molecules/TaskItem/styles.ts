import { StyleSheet } from "react-native";
import { useMetrics } from "@hooks/useMetrics";
import { useThemeStore } from "@store/themeStore";
import { Dimensions } from "@utils";
import { Task } from "src/types/task";

export const useTaskItemStyles = (task: Task) => {
  const { moderateScale, horizontalScale, verticalScale } = useMetrics();
  const { theme } = useThemeStore();

  const containerStyle = {
    backgroundColor: theme.colors.card,
    borderRadius: moderateScale(Dimensions.secondarySpace),
    padding: horizontalScale(Dimensions.sectionSpace),
    marginBottom: verticalScale(Dimensions.secondarySpace),
    borderLeftWidth: moderateScale(Dimensions.tinySpace),
    borderLeftColor: task.completed ? theme.colors.success : theme.colors.primary,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: moderateScale(2),
    },
    shadowOpacity: 0.1,
    shadowRadius: moderateScale(3.84),
    elevation: 5,
  };

  const rowStyle = {
    flexDirection: "row" as const,
    alignItems: "center" as const,
    justifyContent: "space-between" as const,
  };

  const checkboxStyle = {
    marginRight: horizontalScale(Dimensions.secondarySpace),
  };

  const contentStyle = {
    flex: 1,
    marginRight: horizontalScale(Dimensions.secondarySpace),
  };

  const titleStyle = {
    fontSize: moderateScale(Dimensions.regularFont),
    fontWeight: "600" as const,
    color: theme.colors.text,
    textDecorationLine: task.completed ? ("line-through" as const) : ("none" as const),
    opacity: task.completed ? 0.7 : 1,
  };

  const descriptionStyle = {
    fontSize: moderateScale(Dimensions.smallFont),
    color: theme.colors.text,
    opacity: 0.7,
    marginTop: verticalScale(Dimensions.primarySpace),
  };

  const dateStyle = {
    fontSize: moderateScale(Dimensions.smallFont),
    color: theme.colors.text,
    opacity: 0.5,
    marginTop: verticalScale(Dimensions.primarySpace),
  };

  const actionsStyle = {
    flexDirection: "row" as const,
    alignItems: "center" as const,
    gap: horizontalScale(Dimensions.secondarySpace),
  };

  return StyleSheet.create({
    containerStyle,
    rowStyle,
    checkboxStyle,
    contentStyle,
    titleStyle,
    descriptionStyle,
    dateStyle,
    actionsStyle,
  });
};
