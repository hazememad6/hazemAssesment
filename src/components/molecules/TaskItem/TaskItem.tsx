import { Animated, Switch, Text, View } from "react-native";
import React, { memo, useEffect, useRef } from "react";

import { Button } from "@components/atoms";
import { TaskItemProps } from "./types";
import { useTaskItemStyles } from "./styles";

/**
 * TaskItem Component
 *
 * Displays individual task with toggle and delete actions.
 * Using Switch instead of checkbox because it feels more native on mobile.
 *
 * Added smooth animations for completion state:
 * - Fade effect when completing/uncompleting
 * - Scale bounce animation on completion
 * - Smooth opacity transitions for completed state
 *
 * Fixed toggle issue - Switch onValueChange passes boolean value that we need to handle properly.
 */
const TaskItem: React.FC<TaskItemProps> = memo(
  ({ task, onComplete, onDelete, isUpdating = false, isDeleting = false }) => {
    const styles = useTaskItemStyles(task);

    // Animation values
    const fadeAnim = useRef(new Animated.Value(1)).current;
    const scaleAnim = useRef(new Animated.Value(1)).current;
    const opacityAnim = useRef(new Animated.Value(task.completed ? 0.7 : 1)).current;

    // Animate when completion state changes
    useEffect(() => {
      if (task.completed) {
        // Completion animation: scale bounce + fade
        Animated.sequence([
          // Quick scale up
          Animated.timing(scaleAnim, {
            toValue: 1.05,
            duration: 150,
            useNativeDriver: true,
          }),
          // Scale back down with slight overshoot
          Animated.spring(scaleAnim, {
            toValue: 1,
            tension: 300,
            friction: 10,
            useNativeDriver: true,
          }),
        ]).start();

        // Fade to completed state
        Animated.timing(opacityAnim, {
          toValue: 0.7,
          duration: 300,
          useNativeDriver: true,
        }).start();
      } else {
        // Uncomplete animation: just fade back in
        Animated.timing(opacityAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }).start();
      }
    }, [task.completed, scaleAnim, opacityAnim]);

    // Handle completion toggle - receives the new boolean value from Switch
    const handleToggleComplete = (newValue: boolean) => {
      if (isUpdating) return; // prevent toggle during update

      // Only call if the value actually changed to prevent unnecessary calls
      if (newValue !== task.completed) {
        onComplete(task.id);
      }
    };

    const handleDelete = () => {
      if (isDeleting) return; // prevent delete during deletion
      onDelete(task.id);
    };

    return (
      <Animated.View
        style={[
          styles.containerStyle,
          {
            opacity: fadeAnim,
            transform: [{ scale: scaleAnim }],
          },
        ]}
      >
        <View style={styles.rowStyle}>
          <Animated.View style={[styles.contentStyle, { opacity: opacityAnim }]}>
            <Text style={styles.titleStyle} numberOfLines={2}>
              {task.title}
            </Text>
            {task.description && task.description.trim().length > 0 && (
              <Text style={styles.descriptionStyle} numberOfLines={3}>
                {task.description}
              </Text>
            )}
          </Animated.View>

          <View style={styles.actionsStyle}>
            <Switch
              value={task.completed}
              onValueChange={handleToggleComplete}
              disabled={isUpdating} // disable during update
              testID={`task-switch-${task.id}`} // Add testID for tests
              accessibilityLabel={`Mark task ${task.completed ? "incomplete" : "complete"}`}
            />

            <Button
              title="Delete"
              onPress={handleDelete}
              variant="outline"
              size="small"
              disabled={isDeleting} // disable during deletion
              loading={isDeleting} // show loading spinner
              testID={`delete-task-${task.id}`} // Add testID for tests
            />
          </View>
        </View>
      </Animated.View>
    );
  }
);

TaskItem.displayName = "TaskItem";

export { TaskItem };
