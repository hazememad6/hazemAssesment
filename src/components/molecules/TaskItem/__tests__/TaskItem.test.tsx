import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { Task } from "src/types/task";
import { TaskItem } from "../TaskItem";

describe("TaskItem Component", () => {
  const mockTask: Task = {
    id: "1",
    title: "Test Task",
    description: "Test Description",
    completed: false,
    createdAt: "2024-01-01T00:00:00.000Z",
    updatedAt: "2024-01-01T00:00:00.000Z",
  };

  const defaultProps = {
    task: mockTask,
    onComplete: jest.fn(),
    onDelete: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe("Snapshots", () => {
    it("renders incomplete task correctly", () => {
      const tree = render(<TaskItem {...defaultProps} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders completed task correctly", () => {
      const completedTask = { ...mockTask, completed: true };
      const tree = render(<TaskItem {...defaultProps} task={completedTask} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders task without description correctly", () => {
      const taskWithoutDescription = { ...mockTask, description: undefined };
      const tree = render(<TaskItem {...defaultProps} task={taskWithoutDescription} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders task with long title correctly", () => {
      const taskWithLongTitle = {
        ...mockTask,
        title: "This is a very long task title that should wrap properly in the UI",
      };
      const tree = render(<TaskItem {...defaultProps} task={taskWithLongTitle} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Behavior", () => {
    it("displays task title and description", () => {
      const { getByText } = render(<TaskItem {...defaultProps} />);

      expect(getByText("Test Task")).toBeTruthy();
      expect(getByText("Test Description")).toBeTruthy();
    });

    it("calls onToggleComplete when switch is pressed", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const switchComponent = getByTestId("task-switch-1");

      fireEvent(switchComponent, "valueChange", true);

      expect(defaultProps.onComplete).toHaveBeenCalledWith("1");
    });

    it("calls onDelete when delete button is pressed", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const deleteButton = getByTestId("delete-task-1");

      fireEvent.press(deleteButton);

      expect(defaultProps.onDelete).toHaveBeenCalledWith("1");
    });

    it("shows correct switch state for incomplete task", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const switchComponent = getByTestId("task-switch-1");

      expect(switchComponent.props.value).toBe(false);
    });

    it("shows correct switch state for completed task", () => {
      const completedTask = { ...mockTask, completed: true };
      const { getByTestId } = render(<TaskItem {...defaultProps} task={completedTask} />);
      const switchComponent = getByTestId("task-switch-1");

      expect(switchComponent.props.value).toBe(true);
    });

    it("does not render description when not provided", () => {
      const taskWithoutDescription = { ...mockTask, description: undefined };
      const { queryByText } = render(<TaskItem {...defaultProps} task={taskWithoutDescription} />);

      expect(queryByText("Test Description")).toBeNull();
    });
  });

  describe("Accessibility", () => {
    it("has accessible switch component", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const switchComponent = getByTestId("task-switch-1");

      expect(switchComponent).toBeTruthy();
      expect(switchComponent.props.accessibilityRole).toBe("switch");
    });

    it("has accessible delete button", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const deleteButton = getByTestId("delete-task-1");

      expect(deleteButton).toBeTruthy();
      expect(deleteButton.props.accessible).toBe(true);
    });
  });

  describe("Edge Cases", () => {
    it("handles task with empty title", () => {
      const taskWithEmptyTitle = { ...mockTask, title: "" };
      const result = render(<TaskItem {...defaultProps} task={taskWithEmptyTitle} />);

      expect(result).toBeTruthy();
    });

    it("handles task with empty description", () => {
      const taskWithEmptyDescription = { ...mockTask, description: "" };
      const result = render(<TaskItem {...defaultProps} task={taskWithEmptyDescription} />);

      expect(result).toBeTruthy();
    });

    it("handles multiple rapid toggle presses", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const switchComponent = getByTestId("task-switch-1");

      fireEvent(switchComponent, "valueChange", true);
      fireEvent(switchComponent, "valueChange", false);
      fireEvent(switchComponent, "valueChange", true);

      expect(defaultProps.onComplete).toHaveBeenCalledTimes(3);
    });

    it("handles multiple rapid delete presses", () => {
      const { getByTestId } = render(<TaskItem {...defaultProps} />);
      const deleteButton = getByTestId("delete-task-1");

      fireEvent.press(deleteButton);
      fireEvent.press(deleteButton);

      expect(defaultProps.onDelete).toHaveBeenCalledTimes(2);
    });
  });
});
