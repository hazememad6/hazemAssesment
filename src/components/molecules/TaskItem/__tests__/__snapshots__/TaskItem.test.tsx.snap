// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem Component Snapshots renders completed task correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#F2F2F7",
      "borderLeftColor": "#34C759",
      "borderLeftWidth": 2,
      "borderRadius": 8,
      "elevation": 5,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 24,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3.84,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "transformOrigin": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "marginRight": 8,
          "opacity": 0.7,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "fontWeight": "600",
            "textDecorationLine": "line-through",
          }
        }
      >
        Test Task
      </Text>
      <Text
        numberOfLines={3}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "marginTop": 16,
            "opacity": 0.7,
          }
        }
      >
        Test Description
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <RCTSwitch
        accessibilityLabel="Mark task incomplete"
        accessibilityRole="switch"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="task-switch-1"
        value={true}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#007AFF",
            "borderRadius": undefined,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
          }
        }
        testID="delete-task-1"
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": undefined,
              "fontWeight": "600",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TaskItem Component Snapshots renders incomplete task correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#F2F2F7",
      "borderLeftColor": "#007AFF",
      "borderLeftWidth": 2,
      "borderRadius": 8,
      "elevation": 5,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 24,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3.84,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "transformOrigin": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "fontWeight": "600",
            "textDecorationLine": "none",
          }
        }
      >
        Test Task
      </Text>
      <Text
        numberOfLines={3}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "marginTop": 16,
            "opacity": 0.7,
          }
        }
      >
        Test Description
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <RCTSwitch
        accessibilityLabel="Mark task complete"
        accessibilityRole="switch"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="task-switch-1"
        value={false}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#007AFF",
            "borderRadius": undefined,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
          }
        }
        testID="delete-task-1"
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": undefined,
              "fontWeight": "600",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TaskItem Component Snapshots renders task with long title correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#F2F2F7",
      "borderLeftColor": "#007AFF",
      "borderLeftWidth": 2,
      "borderRadius": 8,
      "elevation": 5,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 24,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3.84,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "transformOrigin": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "fontWeight": "600",
            "textDecorationLine": "none",
          }
        }
      >
        This is a very long task title that should wrap properly in the UI
      </Text>
      <Text
        numberOfLines={3}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "marginTop": 16,
            "opacity": 0.7,
          }
        }
      >
        Test Description
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <RCTSwitch
        accessibilityLabel="Mark task complete"
        accessibilityRole="switch"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="task-switch-1"
        value={false}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#007AFF",
            "borderRadius": undefined,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
          }
        }
        testID="delete-task-1"
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": undefined,
              "fontWeight": "600",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TaskItem Component Snapshots renders task without description correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#F2F2F7",
      "borderLeftColor": "#007AFF",
      "borderLeftWidth": 2,
      "borderRadius": 8,
      "elevation": 5,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 24,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3.84,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "transformOrigin": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#000000",
            "fontSize": undefined,
            "fontWeight": "600",
            "textDecorationLine": "none",
          }
        }
      >
        Test Task
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <RCTSwitch
        accessibilityLabel="Mark task complete"
        accessibilityRole="switch"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="task-switch-1"
        value={false}
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#007AFF",
            "borderRadius": undefined,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 16,
          }
        }
        testID="delete-task-1"
      >
        <Text
          style={
            {
              "color": "#007AFF",
              "fontSize": undefined,
              "fontWeight": "600",
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
